keyvi/include/keyvi/dictionary/completion/forward_backward_completion.h

```diff
@@ -236,10 +236,10 @@ class ForwardBackwardCompletion final {
         std::pop_heap(data->results.begin(), data->results.end(), result_compare());
 
         // de-duplicate
-        while (data->last_result.GetMatchedString() == data->results.back().GetMatchedString()) {
+        while (data->last_result && data->last_result->GetMatchedString() == data->results.back()->GetMatchedString()) {
```

Comment:
you have slightly different conventions: [here](https://github.com/KeyviDev/keyvi/pull/298/files#diff-1dbc69e2c71a42075572ad6951771eecba3437f9ab9f4afd3168cc2f7dd8f7f1R150) you explicitly check for `== nullptr` but now you're just using `operator bool` for `last_result`. Since `obj` in the first case and `last_result` in this case are both of type `match_t`, I'm not sure whether there's a reason you're doing it different ways.